source/platforms/github/GitHubGit.ts

```diff
@@ -138,22 +138,33 @@ export default async function gitDSLForGitHub(api: GitHubAPI): Promise<GitDSL> {
 /**
  * Gets the git-style diff for a single file.
  *
- * @param name File path for the diff
+ * @param filename File path for the diff
  */
-  const diffForFile = (name: string, diffTypes?: string[]) => {
-    const diff = find(fileDiffs, (diff: any) => diff.from === name || diff.to === name)
-    if (!diff) { return null }
+  const diffForFile = async (filename: string, diffTypes?: string[]) => {
+    const structuredDiff = find(fileDiffs, (diff: any) => diff.from === filename || diff.to === filename)
+    if (!structuredDiff) { return null }
 
-    let changes = diff.chunks.map((c: any) => c.changes)
-                             .reduce((a: any, b: any) => a.concat(b), [])
+    let changes = structuredDiff.chunks.map((c: any) => c.changes)
+                                .reduce((a: any, b: any) => a.concat(b), [])
 
     // Filter the diffs by type
     if (diffTypes && diffTypes.length > 0) {
       changes = changes.filter((c: any) => diffTypes.indexOf(c.type) !== -1)
     }
 
+    const baseFile = await api.fileContents(filename, pr.base.repo.full_name, pr.base.sha)
```

Comment:
Note: I'm using `sha` here instead of `ref`, mainly because the example PR used for the fixture seemed to have deleted the original branch.

We may also want to use `sha` above in the implementation of `JSONPatchForFile`.